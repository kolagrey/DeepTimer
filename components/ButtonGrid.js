import React from 'react';
import { View } from 'react-native';
import styles from '../styles';

const ButtonGrid = (props) => {
    return(
        <View style={styles.speedControlButtonArea}>
            { props.children }
        </View>
    );
};

export default ButtonGrid;