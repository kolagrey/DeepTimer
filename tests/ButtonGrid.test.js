import React from 'react';
import { Text } from 'react-native';
import renderer from 'react-test-renderer';

import ButtonGrid from '../components/ButtonGrid';


describe('<ButtonGrid/>', () => {

    test('renders correctly', () => {
        const tree = renderer.create(<ButtonGrid/>).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it("is of type 'View'", () => {
        const tree = renderer
            .create(
                <ButtonGrid />,
            ).toJSON();
        expect(tree.type).toBe('View');
    });

    it('can hold a child', () => {
        const tree = renderer
            .create(
                <ButtonGrid>
                    <Text>
                        Hello Test
                    </Text>
                </ButtonGrid>,
            ).toJSON();
        expect(tree.children.length).toBe(1);
    });
});