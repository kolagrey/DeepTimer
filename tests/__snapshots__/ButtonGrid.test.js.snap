// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ButtonGrid/> renders correctly 1`] = `
<View
  style={
    Object {
      "margin": 8,
      "width": 100,
    }
  }
/>
`;
